### PR TITLE
JPN-454 Refactor the Recent Activity Module and add tracking to it

### PR DESCRIPTION
**Pull request by adamkarminski**
### Links
- https://wikia-inc.atlassian.net/browse/JPN-454
### Reviewers

@Wikia/spitfires @perjg @d34th4ck3r

### CODE REVIEW

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialRecentActivityModel.class.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			'moreActivityText' => wfMessage( 'communitypage-recent-activity' )->plain(),
+			'moreActivityLink' => $title->getCanonicalURL(),
+			'activity' => $recentActivity,
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getRecentActivity() {
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialRecentActivityModel.class.php]: Does this method need to be public? Or exist at all?

**Comment by kamilkoterba** [comment]
Have you thought about unit testing it?

File: extensions/wikia/CommunityPage/models/CommunityPageSpecialRecentActivityModel.class.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	/**
+	 * Returns a modified string specifying a type of activity done (create, delete, edit)
+	 * ready to be used in the whole change message.
+	 * @param string $changeType
+	 * @return mixed
+	 */
+	public function getChangeTypeMessage( $changeType ) {
+		switch ( $changeType ) {
+			case 'new':
+				return $changeTypeString = wfMessage( 'communitypage-created' )->plain();
```
Comment (Grunny) [inline extensions/wikia/CommunityPage/models/CommunityPageSpecialRecentActivityModel.class.php]: Left over variable assignment.

**Comment by adamkarminski** [comment]
@kamilkoterba https://wikia-inc.atlassian.net/browse/JPN-509 :)

**Comment by kamilkoterba** [comment]
Code LGTM.
Testing passed.

**Pull request merged**